src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenClosureLambdaTests.cs

```diff
@@ -5224,5 +5224,75 @@ public object Foo<T>()
 }";
             CompileAndVerify(source, new[] { SystemCoreRef });
         }
+
+        [WorkItem(9131, "https://github.com/dotnet/roslyn/issues/9131")]
+        [Fact]
+        public void ClosureInSwitchStatementWithNullableExpression()
+        {
+            string source =
+@"using System;
+class C
+{
+    static void Main()
+    {
+        int? i = null;
+        switch (i)
+        {
+        default:
+            object o = null;
+            Func<object> f = () => o;
+            break;
+        case 0:
+            o = 1;
+            break;
+        }
+    }
+}";
+            var options = TestOptions.DebugExe.WithConcurrentBuild(false);
+            var compilation = CompileAndVerify(source, options: options, expectedOutput: @"");
```

Comment:
Consider printing something to verify that the default section is executed? Also, it wold be good to invoke the lambda and be able to observe the fact of lifting for the local.